src/main/java/com/exasol/adapter/dialects/BaseQueryRewriter.java

```diff
@@ -13,25 +13,21 @@
 /**
  * Base implementation of {@link QueryRewriter}.
  */
-public class BaseQueryRewriter implements QueryRewriter {
+public abstract class BaseQueryRewriter implements QueryRewriter {
```

Comment:
```suggestion
public abstract class AbstractQueryRewriter implements QueryRewriter {
```

The "Base"-part makes no sense anymore if it can't be instantiated.